clang/test/SemaSYCL/attr-add-ir-attributes-function-conflict.cpp

```diff
@@ -0,0 +1,33 @@
+// RUN: %clang_cc1 -fsycl-is-device -fsyntax-only -verify %s
+
+// Tests that add_ir_attributes_function causes a warning when appearing with
+// potentially conflicting SYCL attributes.
+
+constexpr const char AttrName1[] = "Attr1";
+constexpr const char AttrVal1[] = "Val1";
+
+template <const char *... Strs> struct Wrapper {
+  template <typename KernelName, typename KernelType>
+  [[__sycl_detail__::add_ir_attributes_function(Strs...)]] __attribute__((sycl_kernel)) void kernel_single_task(const KernelType &kernelFunc) {
+    kernelFunc();
+  }
+};
+
+int main() {
+  Wrapper<> EmptyWrapper;
+  Wrapper<AttrName1, AttrVal1> NonemptyWrapper;
+
+  EmptyWrapper.kernel_single_task<class EK1>([]() [[sycl::reqd_work_group_size(1)]] {});
+  EmptyWrapper.kernel_single_task<class EK2>([]() [[sycl::reqd_work_group_size(1,2)]] {});
+  EmptyWrapper.kernel_single_task<class EK3>([]() [[sycl::reqd_work_group_size(1,2,3)]] {});
+  EmptyWrapper.kernel_single_task<class EK7>([]() [[sycl::reqd_sub_group_size(1)]] {});
```

Comment:
The test case for `work_group_size_hint` attribute seems to be missing.

Reply:
There is currently a bug with the placement of `[[sycl::work_group_size_hint]]` preventing it from being like these. It may be possible to use it in another location. I will have a look.

Reply:
I think it should be fixed by https://github.com/intel/llvm/pull/6976 .

Reply:
I have added case for `sycl::work_group_size_hint` as well, thanks!